### PR TITLE
Fix: update stravalib typing to Python 3.10+ using pyupgrade + cleanup

### LINKED ISSUE #547: Upgrade typing to python 3.10
Given we have dropped python 3.9, i am going to upgrade all modules to 3.10+ using [pyupgrade](https://github.com/asottile/pyupgrade). This should be a quick easy pr to create and thus can be included in the stravalib 2.x release.

### PR DESCRIPTION
**Pull request by lwasser**
closes #547 

This is an upgrade pr that removes Optional and Union in favor of | and None

## Type of change

Select the statement best describes this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [ ] This is a infrastructure update (docs, ci, etc)
- [x] Other (please describe) just a typing upgrade

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [x] This change doesn't require tests

## Did you include your contribution to the change log?

- [X] Yes, `changelog.md` is up-to-date.

### CODE REVIEW

File: src/stravalib/model.py [hunk 1]
```diff
@@ -14,20 +14,16 @@
 
 import logging
 from abc import ABC, abstractmethod
-from collections.abc import Sequence
+from collections.abc import Callable, Mapping, Sequence
```
Comment (lwasser) [inline src/stravalib/model.py:17]: oh this is interesting. was Callable  & Mapping moved to `collections.abc` ?

File: src/stravalib/protocol.py [hunk 2]
```diff
@@ -364,10 +365,10 @@ def _handle_protocol_error(
 
         # Special subclasses for some errors
         if response.status_code == 404:
-            msg = "{}: {}".format(response.reason, error_str)
+            msg = f"{response.reason}: {error_str}"
```
Comment (lwasser) [inline src/stravalib/protocol.py:368]: This is neat that it knows to upgrade to f-strings too!

File: src/stravalib/strava_model.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from __future__ import annotations
```
Comment (lwasser) [inline src/stravalib/strava_model.py:4]: I think we will want to update the [other repo here](https://github.com/stravalib/strava_swagger2pydantic/blob/main/swagger2pydantic.py#L87) to support this. i'll leave this in draft until i actually do that and regenerate the model from that script

**Review by lwasser — COMMENTED** [review]
Ok this is not quite ready for review. 

I forgot about the model generation repo. i'll open that pr later and will properly regenerate strava_model from there to ensure both repos are fully synced!

**Comment by lwasser** [comment]
ok this is ready for review - and it can be paired with [this pr](https://github.com/stravalib/strava_swagger2pydantic/pulls) that updates the model generator! i did a bit of file cleanup in this pr as well. 

note: we could in the future also drop black and just use ruff format to further streamline our builds. I'll open a pr for this later once we have gotten everything we need to stravalib 2.x pulled together and merged / released!

File: src/stravalib/strava_model.py (hunk 3 above)
Comment (lwasser) [reply src/stravalib/strava_model.py:4]: this is done!

File: src/stravalib/strava_model.py
Comment (lwasser) [inline src/stravalib/strava_model.py:1]: this is a generated file now!

**Comment by lwasser** [comment]
hey @jsamoocha what do you think about this pr? i'm wondering if we can merge it and then consider merging and releasing stravalib 2.0 now?

**Review by jsamoocha — APPROVED** [review]
Thanks, this looks good. Sorry for not looking into it earlier - I kinda missed these pending PRs

**Comment by lwasser** [comment]
Please don't apologize!! I'm just catching up on things and given the recent issue posted, I was thinking it might be good to nudge the new release forward! Thank you!

**Pull request merged**